### DISCUSSION TITLE
using nx with commander app

### DISCUSSION THREAD

**Discussion by coffeebe4code** [nrwl/nx#17159] [General]
I didn't see a generator for a node cli app.

Almost all of my projects are this.

I wanted to ask the above question first, to prevent the xy problem.

But, I am to the point where it builds and packages it, and i can npm install the directory making the resulting cli-app.js file a binary. However, my other library ends up becoming a peer dependency. I would rather all my node cli apps, not have peer dependencies.

So first I ask if there is a supported way to do cli apps. 

Next I ask, is how do i prevent the peer dependency issue from happening in a command line app.

after installing my cli app.

```
npm WARN @organization/cli-app@1.0.1 requires a peer of @organization/logging-package@0.0.1 but none is installed. You must install peer dependencies yourself.
npm WARN @organization/cli-app@1.0.1 requires a peer of tslib@2.5.2 but none is installed. You must install peer dependencies yourself.

+ @organization/cli-app@1.0.1
```

**Comment by hanspoo** [comment]
Actually i modified the serve target in project.json; for example for an integrated monorepo with a node app called cli:

```
    "serve": {
      "executor": "nx:run-commands",
      "defaultConfiguration": "development",
      "options": {
        "command": "nx run cli:build; node dist/apps/cli/main.js"
      }
    }
```

For example using commander first sample without any modification in main.ts, you can call:

``nx serve cli -- split --separator=/ a/b/c``
